crates/apub/src/objects/note_wrapper.rs

```diff
@@ -0,0 +1,85 @@
+use super::comment::ApubComment;
+use crate::{
+  objects::private_message::ApubPrivateMessage,
+  protocol::objects::note_wrapper::NoteWrapper,
+};
+use activitypub_federation::{config::Data, kinds::public, traits::Object};
+use chrono::{DateTime, Utc};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_utils::error::{LemmyError, LemmyResult};
+use serde_json::{from_value, to_value};
+use url::Url;
+
+/// Private messages and public comments are quite awkward in Activitypub, because the json
+/// format looks identical. They only way to differentiate them is to check for the presence
+/// or absence of `https://www.w3.org/ns/activitystreams#Public` in `to` or `cc` which this
+/// wrapper does.
+#[derive(Debug)]
+pub(crate) struct ApubNote {}
+
+#[async_trait::async_trait]
+impl Object for ApubNote {
+  type DataType = LemmyContext;
+  type Kind = NoteWrapper;
+  type Error = LemmyError;
+
+  fn last_refreshed_at(&self) -> Option<DateTime<Utc>> {
+    None
+  }
+
+  #[tracing::instrument(skip_all)]
+  async fn read_from_id(
+    _object_id: Url,
+    _context: &Data<Self::DataType>,
+  ) -> LemmyResult<Option<Self>> {
+    unimplemented!()
```

Comment:
This will pry fail CI, since we have a clippy deny for that.

Reply:
Changed these to errors